fix(core): gonka_tests use super::* misses build_provider_from_entry — unit tests fail to compile with --features full

## Description

PR #3632 introduced 4 new unit tests in `crates/zeph-core/src/provider_factory.rs` inside a nested `#[cfg(feature = \"gonka\")] mod gonka_tests`. These tests call `build_provider_from_entry` via `use super::*`, but `super` resolves to `mod tests` — which only imports `effective_embedding_model` and `stable_skill_embedding_model`, not `build_provider_from_entry`. As a result, `cargo nextest run --workspace --features full --lib --bins` fails to compile `zeph-core`.

## Reproduction Steps

1. `git checkout b0a07602`
2. `cargo nextest run --config-file .github/nextest.toml --workspace --features full --lib --bins`
3. Observe 4 errors: `cannot find function build_provider_from_entry in this scope`

## Expected Behavior

All tests compile and pass.

## Actual Behavior

```
error[E0425]: cannot find function `build_provider_from_entry` in this scope
  --> crates/zeph-core/src/provider_factory.rs:633:26
  --> crates/zeph-core/src/provider_factory.rs:647:26
  --> crates/zeph-core/src/provider_factory.rs:663:26
  --> crates/zeph-core/src/provider_factory.rs:677:26
error: could not compile `zeph-core` (lib test) due to 4 previous errors
```

## Fix

In `mod tests`, add:
```rust
use super::build_provider_from_entry;
```

Or move the `use` into `gonka_tests` directly:
```rust
use super::super::build_provider_from_entry;
```

## Environment

- Commit: b0a07602
- Features: full (gonka enabled)

## Logs / Evidence

Detected in CI-687 live testing pre-flight (cargo nextest run step).

--- a/crates/zeph-core/src/provider_factory.rs
+++ b/crates/zeph-core/src/provider_factory.rs
@@ -594,7 +594,9 @@ mod tests {
         ));
     }
 
-    use super::{effective_embedding_model, stable_skill_embedding_model};
+    use super::{
+        build_provider_from_entry, effective_embedding_model, stable_skill_embedding_model,
+    };
     use crate::config::{Config, ProviderKind};
     use zeph_config::providers::ProviderEntry;
 

--- a/crates/zeph-core/src/provider_factory.rs
+++ b/crates/zeph-core/src/provider_factory.rs
@@ -594,9 +594,9 @@ mod tests {
         ));
     }
 
-    use super::{
-        build_provider_from_entry, effective_embedding_model, stable_skill_embedding_model,
-    };
+    #[cfg(feature = "gonka")]
+    use super::build_provider_from_entry;
+    use super::{effective_embedding_model, stable_skill_embedding_model};
     use crate::config::{Config, ProviderKind};
     use zeph_config::providers::ProviderEntry;
 

